File: Sem2/PO_projekt/Java/plants/Grass.java
package Java.plants;

import Java.Game;
import Java.Organism;
import Java.Plant;
import java.awt.Color;

public class Grass extends Plant {
    public Grass(int x, int y, Game game) {
        super(0,0, x, y, game);
    }

    @Override
    public Color getColor() {
        return new Color(0, 255, 0);
    }

    @Override
    public boolean ignoredByPredator() {
        return true;
    }

    @Override
    public Organism getOrganism() {
        return this;
    }

    @Override
    protected Plant spread(int targetX, int targetY) {
        return new Grass(targetX, targetY, game);
    }
}

File: Sem2/PO_project/python/plants/grass.py
from __future__ import annotations

from python.organism import Plant


class Grass(Plant):
    def __init__(self, x: int, y: int, game) -> None:
        super().__init__(0, 0, x, y, game)

    def get_color(self) -> tuple[int, int, int]:
        return (0, 255, 0)

    def ignored_by_predator(self) -> bool:
        return True

    def get_organism(self):
        return self

    def spread(self, target_x: int, target_y: int):
        return Grass(target_x, target_y, self.game)
